ParseError does not work well with anyhow

The following code fails to compile:
```rs
use anyhow::Error;
use prse::{try_parse, ParseError};

fn main() -> Result<(), Error> {
    let l = "test: 5";

    let y = try_parse!(l, "test: {}")?;

    assert_eq!(5_i32, y);
    Ok(())
}
```
I get the following error:
```
error[E0277]: `(dyn std::error::Error + 'static)` cannot be sent between threads safely
 --> examples/main.rs:7:38
  |
7 |     let y = try_parse!(l, "test: {}")?;
  |                                      ^ `(dyn std::error::Error + 'static)` cannot be sent between threads safely
  |
  = help: the trait `Send` is not implemented for `(dyn std::error::Error + 'static)`
  = help: the following other types implement trait `FromResidual<R>`:
            <Result<T, F> as FromResidual<Result<Infallible, E>>>
            <Result<T, F> as FromResidual<Yeet<E>>>
  = note: required for `Unique<(dyn std::error::Error + 'static)>` to implement `Send`
  = note: required because it appears within the type `Box<(dyn std::error::Error + 'static)>`
  = note: required because it appears within the type `ParseError`
  = note: required for `anyhow::Error` to implement `From<ParseError>`
  = note: required for `Result<(), anyhow::Error>` to implement `FromResidual<Result<Infallible, ParseError>>`

error[E0277]: `(dyn std::error::Error + 'static)` cannot be shared between threads safely
 --> examples/main.rs:7:38
  |
7 |     let y = try_parse!(l, "test: {}")?;
  |                                      ^ `(dyn std::error::Error + 'static)` cannot be shared between threads safely
  |
  = help: the trait `Sync` is not implemented for `(dyn std::error::Error + 'static)`
  = help: the following other types implement trait `FromResidual<R>`:
            <Result<T, F> as FromResidual<Result<Infallible, E>>>
            <Result<T, F> as FromResidual<Yeet<E>>>
  = note: required for `Unique<(dyn std::error::Error + 'static)>` to implement `Sync`
  = note: required because it appears within the type `Box<(dyn std::error::Error + 'static)>`
  = note: required because it appears within the type `ParseError`
  = note: required for `anyhow::Error` to implement `From<ParseError>`
  = note: required for `Result<(), anyhow::Error>` to implement `FromResidual<Result<Infallible, ParseError>>`

error: aborting due to 2 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0277`.
error: could not compile `prse` due to 3 previous errors; 1 warning emitted
Process finished with exit code 101
```

--- a/src/parse_error.rs
+++ b/src/parse_error.rs
@@ -31,7 +31,7 @@ pub enum ParseError {
     /// The variant returned when you want to return an error that is not defined here.
     /// When not using the `std` feature, `Dyn` is a unit variant as the
     /// [`Error`](error::Error) trait is part of std.
-    Dyn(Box<dyn error::Error>),
+    Dyn(Box<dyn error::Error + Send + Sync>),
     /// The variant returned when you want to return an error that is not defined here.
     /// When not using the `std` feature, `Dyn` is a unit variant as the
     /// [`Error`](error::Error) trait is part of std.
@@ -62,15 +62,13 @@ pub enum ParseError {
 #[cfg(feature = "std")]
 impl error::Error for ParseError {
     fn source(&self) -> Option<&(dyn error::Error + 'static)> {
-        use std::borrow::Borrow;
-
         match self {
             ParseError::Int(source) => Some(source),
             ParseError::Bool(source) => Some(source),
             ParseError::Char(source) => Some(source),
             ParseError::Float(source) => Some(source),
             ParseError::Addr(source) => Some(source),
-            ParseError::Dyn(source) => Some(source.borrow()),
+            ParseError::Dyn(source) => Some(&**source),
             ParseError::Literal { .. } => None,
             ParseError::Multi { .. } => None,
         }
@@ -163,4 +161,18 @@ impl_from_parse_error!(ParseFloatError, Float);
 #[cfg(feature = "std")]
 impl_from_parse_error!(AddrParseError, Addr);
 #[cfg(feature = "std")]
-impl_from_parse_error!(Box<dyn error::Error>, Dyn);
+impl_from_parse_error!(Box<dyn error::Error + Send + Sync>, Dyn);
+
+#[cfg(test)]
+mod test {
+    use crate::ParseError;
+
+    #[test]
+    fn check_impl_traits() {
+        fn is_send<T: Send>() {}
+        fn is_sync<T: Sync>() {}
+
+        is_send::<ParseError>();
+        is_sync::<ParseError>();
+    }
+}

